api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -27,16 +38,100 @@ public NameSupport() {
 	}
 	
 	public static NameSupport getInstance() {
-		if (singleton == null) {
-			throw new APIException("Not Yet Instantiated");
-		} else {
-			return singleton;
+		synchronized (NameSupport.class) {
+			singleton.init();
+			if (singleton == null) {
```

Comment:
If this were null, a NPE would have already been thrown above. Not so?

Reply:
Nice catch @dkayiwa...